modules/framework/galasa-parent/dev.galasa.framework.api.streams/src/main/java/dev/galasa/framework/api/streams/internal/routes/StreamsTransform.java

```diff
@@ -0,0 +1,124 @@
+     */
+    public Stream createStreamBean(IStream stream, String apiServerUrl) {
+
+        Stream streamBean = new Stream();
+        streamBean.setApiVersion(GalasaResourceValidator.DEFAULT_API_VERSION);
+        streamBean.setkind("GalasaStream");
+
+        // Build metadata (name, url, description)
+        StreamMetadata streamMetadata = createStreamMetadata(stream, apiServerUrl);
+        streamBean.setmetadata(streamMetadata);
+
+        // Build data section (isEnabled, repository, obrs, testCatalog)
+        StreamData streamData = createStreamData(stream);
+        streamBean.setdata(streamData);
+
+        return streamBean;
+    }
+
+    /**
+     * Creates the metadata section of the Stream bean.
+     * It looks for keys that start with "test.stream." and are not part of the data
+     * section.
+     * For example: "test.stream.mystream.description" and
+     * "test.stream.mystream.url".
+     */
+    private StreamMetadata createStreamMetadata(IStream stream, String apiServerUrl) {
+        logger.info("Building stream metadata");
+
+        String streamName = stream.getName();
+        StreamMetadata metadata = new StreamMetadata();
+        metadata.setname(streamName);
+
+        String streamDescription = stream.getDescription();
+        if (streamDescription != null && !streamDescription.isBlank()) {
+            metadata.setdescription(stream.getDescription());
+        }
+
+        metadata.seturl(apiServerUrl + "/streams/" + streamName);
+        return metadata;
+    }
+
+    /**
+     * Creates the data section of the Stream bean.
+     * It processes keys for isEnabled, repository, obrs and testCatalog.
+     */
+    private StreamData createStreamData(IStream stream) {
+
+        StreamData data = new StreamData();
+        List<StreamOBRData> streamObrData = new ArrayList<>();
+        String testCatalogUrl = stream.getTestCatalogUrl();
+        String obrLocation = stream.getObrLocation();
+
+        StreamRepository streamRepository = new StreamRepository();
+        streamRepository.seturl(stream.getMavenRepositoryUrl());
+
+        StreamTestCatalog testCatalog = new StreamTestCatalog();
+        testCatalog.seturl(testCatalogUrl);
+
+        if (obrLocation != null) {
+            // Strip off the "mvn:" prefix.
+            String[] obrUrls = obrLocation.split(",");
+            streamObrData = transformMultipleObrs(obrUrls);
+        }
+
+        data.setobrs(streamObrData.toArray(new StreamOBRData[0]));
+        data.setTestCatalog(testCatalog);
+        data.setrepository(streamRepository);
+        data.setIsEnabled(true);
+
+        return data;
+
+    }
+
+    private List<StreamOBRData> transformMultipleObrs(String[] obrUrls) {
+
+        List<StreamOBRData> streamObrData = new ArrayList<>();
+
+        for (String obrLocation : obrUrls) {
+            String extractedUrl = obrLocation.substring(4, obrLocation.length() - 1);
```

Comment:
Where does the 4 come from? If it's the `mvn:` prefix, we could use that as a constant and do a `.length()` on it?